### ISSUE TITLE
Calendar Error during PROPFIND on Arch Linux with Thunderbird

### ISSUE THREAD

**Issue by ChasNelson1990** [etesync/etesync-dav#52]
Hi there,

Recently moved to Thunderbird 60.5.3-1 (from Evolution) and loaded my calendars.

As with Evolution (see #16) one calendar works fine. The other fails to sync, but with a different error:
> ERROR: An exception occurred during PROPFIND request on 'MY CALENDAR URL': uid

If I log in to the Radical web server and download the ics, I get:

> BEGIN:VCALENDAR
> VERSION:2.0
> PRODID:-//PYVOBJECT//NONSGML Version 1//EN
> END:VCALENDAR

Despite being able to see 50 events or so in the calendar on Android and in the web app.

Any ideas?

**Comment by tasn** [comment]
Try removing the etesync dav cache:

    rm ~/.config/etesync-dav/etesync_data.db

And then running etesync-dav again.

**Comment by ChasNelson1990** [comment]
Restarted. Got:
> ERROR: An exception occurred during REPORT request on 'email@domain.co.uk/key': uid
On first sync attempt.
Got:
> WARNING: Client provided invalid sync token 'http://radicale.org/ns/sync/SOMEHASH': Sync token are not supported

I've read #35, so don't think the warning should matter. But still no events appear, so the error clearly does.

This is repeatable if I clear the cache and restart again.

**Comment by tasn** [comment]
1. Are you using an up to date etesync-dav?
2. Could you please run it with -D so we get proper debug info?

Thanks!

**Comment by ChasNelson1990** [comment]
Sorry. I'm using the Arch packages:
python-etesync==0.8.1
etesync-dav==0.5.0
python-radicale-storage-etesync==0.7.0
Radicale==2.1.11

Full `-D` output:
>INFO: Starting Radicale
>INFO: Authentication type is 'htpasswd'
>INFO: Storage type is 'radicale_storage_etesync'
>INFO: Rights type is 'owner_only'
>INFO: Web type is 'internal'
>INFO: Listening to 'localhost' on port 37358
>INFO: Radicale server ready
>INFO: PROPFIND request for '/email@domain.co.uk/key/' with depth '0' received from 127.0.0.1 using 'Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 Thunderbird/60.5.3 Lightning/6.2.5.3'
>DEBUG: Request headers:
>{'CONTENT_LENGTH': '306',
> 'CONTENT_TYPE': 'text/xml; charset=utf-8',
> 'GATEWAY_INTERFACE': 'CGI/1.1',
> 'HTTP_ACCEPT': 'text/xml',
> 'HTTP_ACCEPT_CHARSET': 'utf-8,*;q=0.1',
> 'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
> 'HTTP_ACCEPT_LANGUAGE': 'en-GB,en;q=0.5',
> 'HTTP_AUTHORIZATION': 'Basic **masked**',
> 'HTTP_CACHE_CONTROL': 'no-cache',
> 'HTTP_CONNECTION': 'keep-alive',
> 'HTTP_DEPTH': '0',
> 'HTTP_HOST': 'localhost:37358',
> 'HTTP_PRAGMA': 'no-cache',
> 'HTTP_USER_AGENT': 'Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 '
                    'Thunderbird/60.5.3 Lightning/6.2.5.3',
> 'PATH_INFO': '/email@domain.co.uk/key/',
> 'QUERY_STRING': '',
> 'REMOTE_ADDR': '127.0.0.1',
> 'REMOTE_HOST': '',
> 'REQUEST_METHOD': 'PROPFIND',
> 'SCRIPT_NAME': '',
> 'SERVER_NAME': 'localhost',
> 'SERVER_PORT': '37358',
> 'SERVER_PROTOCOL': 'HTTP/1.1',
> 'SERVER_SOFTWARE': 'WSGIServer/0.2',
> 'wsgi.errors': <_io.StringIO object at 0x7f09c8f90ca8>,
> 'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
> 'wsgi.input': <_io.BufferedReader name=6>,
> 'wsgi.multiprocess': False,
> 'wsgi.multithread': True,
> 'wsgi.run_once': False,
> 'wsgi.url_scheme': 'http',
> 'wsgi.version': (1, 0)}
>DEBUG: Sanitized script name: ''
>DEBUG: Sanitized path: '/email@domain.co.uk/key/'
>INFO: Successful login: 'email@domain.co.uk'
>ERROR: An exception occurred during PROPFIND request on '/email@domain.co.uk/key/': uid
>Traceback (most recent call last):
>  File "/usr/lib/python3.7/site-packages/vobject/base.py", line 529, in __getattr__
>    return self.contents[toVName(name)][0]
>KeyError: 'uid'
>
>During handling of the above exception, another exception occurred:
>
>Traceback (most recent call last):
>  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 332, in __call__
>    status, headers, answers = self._handle_request(environ)
>  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 477, in _handle_request
>    with self.Collection.acquire_lock("r", user):
>  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
>    return next(self.gen)
>  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 514, in acquire_lock
>    cls.etesync.pull_journal(journal.uid)
>  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 200, in pull_journal
    collection.apply_sync_entry(syncEntry)
>  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 406, in apply_sync_entry
    uid = self.get_content_class().get_uid(sync_entry.content)
>  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 366, in get_uid
    return vobj.uid.value
>  File "/usr/lib/python3.7/site-packages/vobject/base.py", line 531, in __getattr__
    raise AttributeError(name)
>AttributeError: uid

**Comment by tasn** [comment]
OK, it looks like you have a bad entry in your etesync journal that etesync-dav doesn't know how to deal with.

Let's edit the code to add some debug prints. Please open `/usr/lib/python3.7/site-packages/etesync/api.py` and change line 406 (`uid = self.get_content_class().get_uid(sync_entry.content)`) to:

```
try:
    uid = self.get_content_class().get_uid(sync_entry.content)
except Exception as e:
    print(sync_entry.content)
    print(self.__class__.TYPE)
    raise e
```

The output will contain the content of your task, so feel free to censor it a bit.

It's in your tasks btw. I wonder how a bad entry got in there. Though since it's been broken for a while for you, maybe it's an old thing that's now been fixed and you're just seeing the leftover of that.

**Comment by tasn** [comment]
I made a few improvements and fixes to etesync-dav, could you please try updating this and the underlying packages (radicale-storage-etesync and etesync) and see if that fixes it for you?

**Comment by ChasNelson1990** [comment]
Hi, when I did an AUR update (using pikaur) I got the following message:

> New packages 'etesync-dav' and 'python-radicale-storage-etesync' are in conflict.

So I uninstalled me existing etesync-dav (and dependencies, including radicale; again using pikaur).
I then, successfully, reinstalled etesync-dav.

However, running `etesync-dav` gives me the following:

> Traceback (most recent call last):
>     File "/usr/bin/etesync-dav", line 6, in <module>
>         import radicale.__main__
> ModuleNotFoundError: No module named 'radicale'

I then manually installed radicale and now get:

> ERROR: An exception occurred during server startup: Failed to load storage module 'radicale_storage_etesync': No module named 'etesync'

So I looked at manually installing `python-radicale-storage-etesync` and I get told there's a conflict again.

Based on your commit messages, this should all be merged now so that I don't need to install `python-radicale-storage-etesync`, right?

**Comment by tasn** [comment]
Sorry, fixed now. The package was missing the radicale-storage-etesync dependencies. I'm at a place with a really bad connection so was being a bit sloppy with the testing.

As for your question, yeah radicale-storage-etesync is now deprecated and unneeded.

**Comment by ChasNelson1990** [comment]
Install and run all seem fine now - thanks.

I had, for other reasons, no longer got these calendars installed on this PC but I reinstalled them using the full collection path.

So I have one calendar that used to work fine, and one that had this issue.

Both keep greying themselves out in Thunderbird. I have to right click > 'Properties' > 'Switch this calendar on'. And then as soon as I hit 'Synchronise' it disables itself again.

**Comment by tasn** [comment]
Are you able to download the .ics from the web UI without a problem? Does Thunderbird report anything in particular? Is there anything interesting when you run etesync-dav with -D?

**Comment by ChasNelson1990** [comment]
The debug isn't outputting anything unusual... however, the .ics files are empty if I download them. Thunderbird reports nothing.

This is the debug for an .ics download (hash looking things removed):

> [7fb6fa7f5700] INFO: GET request for '<REMOVED>' received from 127.0.0.1 using 'Mozilla/5.0 (X11; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0'
[7fb6fa7f5700] DEBUG: Request headers:
{'CONTENT_LENGTH': '',
 'CONTENT_TYPE': 'text/plain',
 'GATEWAY_INTERFACE': 'CGI/1.1',
 'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
 'HTTP_ACCEPT_LANGUAGE': 'en-GB,en;q=0.5',
 'HTTP_AUTHORIZATION': 'Basic **masked**',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_COOKIE': '**masked**',
 'HTTP_DNT': '1',
 'HTTP_HOST': 'localhost:37358',
 'HTTP_IF_MODIFIED_SINCE': 'Fri, 12 Jul 2019 12:02:14 GMT',
 'HTTP_IF_NONE_MATCH': '<REMOVED>',
 'HTTP_REFERER': 'http://localhost:37358/.web/',
 'HTTP_UPGRADE_INSECURE_REQUESTS': '1',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (X11; Linux x86_64; rv:67.0) Gecko/20100101 '
                    'Firefox/67.0',
 'PATH_INFO': '<REMOVED>',
 'QUERY_STRING': '',
 'REMOTE_ADDR': '127.0.0.1',
 'REMOTE_HOST': '',
 'REQUEST_METHOD': 'GET',
 'SCRIPT_NAME': '',
 'SERVER_NAME': 'localhost',
 'SERVER_PORT': '37358',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'SERVER_SOFTWARE': 'WSGIServer/0.2',
 'wsgi.errors': <_io.StringIO object at 0x7fb6fcbcb438>,
 'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
 'wsgi.input': <_io.BufferedReader name=6>,
 'wsgi.multiprocess': False,
 'wsgi.multithread': True,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}
[7fb6fa7f5700] DEBUG: Sanitized script name: ''
[7fb6fa7f5700] DEBUG: Sanitized path: '<REMOVED>'
[7fb6fa7f5700] INFO: Successful login: '<REMOVED>'
[7fb6fa7f5700] INFO: Collection read 0 items in 0.000375 sec from <REMOVED>
[7fb6fa7f5700] DEBUG: Response content:
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//PYVOBJECT//NONSGML Version 1//EN
END:VCALENDAR
[7fb6fa7f5700] INFO: GET response status for '<REMOVED>' in 0.023 seconds: 200 OK

**Comment by tasn** [comment]
Have you tried deleting your cache and starting over as suggested above?
`rm ~/.config/etesync-dav/etesync_data.db`

It looks like something is wrong with your setup (though it should have at least printed something to stdout). Do you maybe have a lot of data on your account? (it should still work, but I'm just trying to narrow it down by finding out what may be unusual in your case).

**Comment by ChasNelson1990** [comment]
I'll do the cache thing now. Is there a way to work out my account data usage? I don't believe it's a lot?

**Comment by tasn** [comment]
You can check the amount of entries each of your journals have in the Android app.

**Comment by ChasNelson1990** [comment]
345 and 283 entries in the journals I'm working with right now.

Cleared the cache, restarted, downloaded the .ics still empy.

**Comment by ChasNelson1990** [comment]
I occasionally get:
> ERROR: An exception occurred during GET request on '/': uid
When navigating the web gui, but if I just refresh the page it works fine.

**Comment by tasn** [comment]
Can you get this error with -D? I think it may just be the error you described somewhere above. And essentially you have a improperly constructed event somewhere in the change history which is probably why it's not showing anything.

**Comment by ChasNelson1990** [comment]
Yes, sorry. It is what I reported above (I had another issue at a different time and in my head it was still that one.)

The full debug is:

>[7f01d5c5c700] ERROR: An exception occurred during GET request on '/.web/': uid
Traceback (most recent call last):
  File "/home/chas/.local/lib/python3.7/site-packages/vobject/base.py", line 529, in __getattr__
    return self.contents[toVName(name)][0]
KeyError: 'uid'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 332, in __call__
    status, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 477, in _handle_request
    with self.Collection.acquire_lock("r", user):
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 563, in acquire_lock
    cls.etesync.pull_journal(journal.uid)
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 205, in pull_journal
    collection.apply_sync_entry(syncEntry)
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 411, in apply_sync_entry
    uid = self.get_content_class().get_uid(sync_entry.content)
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 371, in get_uid
    return vobj.uid.value
  File "/home/chas/.local/lib/python3.7/site-packages/vobject/base.py", line 531, in __getattr__
    raise AttributeError(name)
AttributeError: uid

**Comment by ChasNelson1990** [comment]
But is this not just when radicale is trying to load the journal list, before accessing an individual journal?

**Comment by ChasNelson1990** [comment]
Or perhaps `cls.etesync.pull_journal(journal.uid)`pull all the journal data in at once?

**Comment by tasn** [comment]
etesync-dav periodically syncs everything, that's what you're seeing.

It looks like one of your contacts is missing a uid. Could you please add `print(content)` to the line before the return in  `File "/usr/lib/python3.7/site-packages/etesync/api.py", line 371, in get_uid`?

This would show us the broken contact. Then if you can, just change the name, email, phone number and everything that's private to something fake (but don't remove lines) and share that here? I'd love to see the broken contact and this will also enable me to reproduce it locally and test it properly.

Thanks!

**Comment by ChasNelson1990** [comment]
I've changed the name to TASN but otherwise there's not much in there. Seems to be my first contact too.

>BEGIN:VCARD
VERSION:4.0
PRODID:-//EteSync//com.etesync.syncadapter 0.14.0//vcard4android ez-vcard/0.10.x
FN:TASN
N:;TASN;;;
REV:20170510T143030Z
END:VCARD

**Comment by ChasNelson1990** [comment]
There also isn't an entry in my phone contacts app for TASN (forename only).

**Comment by tasn** [comment]
Can you please try replacing line 411 with:
```
        try:
            uid = self.get_content_class().get_uid(sync_entry.content)
        except AttributeError:
            print("WARNING: uid not found for entry, skipping. Content:")
            print(sync_entry.content)
            return
```

A bit hacky, but it should fix this. I'll probably do something slightly different for the released solution, but just want to see if it works.

**Comment by ChasNelson1990** [comment]
Okay... we're getting there. Modification made - both calendars will now sync with Thunderbird.

But! If I try to dismiss a reminder (or rather the 20 it wants to tell me on first sync) the calendar is seemingly set to read-only as Thunderbird gives me a MODIFICATION_FAILED error.

If I reset the read-only check box in Thunderbird it resyncs, I get all the notifications again and then it does read-only when I dismiss them.

There seems to be an associated debug message:

> [7fc59714b700] ERROR: An exception occurred during PUT request on 'REMOVED': unsupported operand type(s) for +: 'UUID' and 'str'
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 332, in __call__
    status, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 505, in _handle_request
    environ, base_prefix, path, user)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 935, in do_PUT
    path, items, props)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 328, in create_collection
    vobject_items.get)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 260, in _find_available_file_name
    file_name = uuid4() + suffix
TypeError: unsupported operand type(s) for +: 'UUID' and 'str'

**Comment by tasn** [comment]
Sorry for all the hot-patching, but can you please also change:
` File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/init.py", line 260, in _find_available_file_name`
to
file_name = str(uuid4()) + suffix`

**Comment by ChasNelson1990** [comment]
That error has no gone but the calendars still revert to read-only and can't actually dismiss a reminder.

The associated error seems to be:

> ERROR: An exception occurred during PUT request on 'REMOVED': 'CollectionCopy' object has no attribute 'upload_all_nonatomic'
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 332, in __call__
    status, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 505, in _handle_request
    environ, base_prefix, path, user)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 935, in do_PUT
    path, items, props)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 330, in create_collection
    self.upload_all_nonatomic(vobject_items)
AttributeError: 'CollectionCopy' object has no attribute 'upload_all_nonatomic'

**Comment by tasn** [comment]
It's so weird you keep on getting these errors. Thunderbird works fine here
and for many users on Thunderbird. This looks like a real error, I wonder
why this path is not tested.

Unfortunately I'm currently away from keyboard and will be so until Monday,
but keep on coming, post here any issue you see and I'll keep on fixing
them until we figure it out if that's ok with you.

--
Tom

**Comment by ChasNelson1990** [comment]
Happy to. I'll be afk in a little while for the rest of the day (UTC+1) but should be around over the weekend to follow any extra changes up.

**Comment by ChasNelson1990** [comment]
Okay, back at my keyboard, on a different Arch Linux with Thunderbird PC.

I can recreate the original error.

I then made the two changes we made above and have recreated the `'CollectionCopy'` error that we got to.

**Comment by tasn** [comment]
I'm also on Arch, and use Android, but I install everything from pip in an
env (and don't have a contact without a uid), but still no errors here.
I'll investigate more on Monday.

--
Tom

**Comment by ChasNelson1990** [comment]
As I think mentioned earlier, I'm installing through AUR.

Also, just for info, I have no issues with the Android app for these journals.

**Comment by tasn** [comment]
I've done a bit more testing and it looks like you either created a calendar from the web UI or from Thunderbird itself, is this correct? Because looking at the code path, it looks like it's what's happening. I tried installing etesync-dav from AUR but everything works great, so I don't think it's something with the AUR packages.

I fixed a few more things (you can use this file to test: https://github.com/etesync/etesync-dav/blob/tbird/radicale_storage_etesync/__init__.py), though the reason why it's broken in the first place, is because this code path should never be called.

I think what's going on is that you've been a long time user, so your etesync-dav (~/.config/etesync-dav/radicale.conf) configuration is missing
```
[rights]
type = radicale_storage_etesync.rights
```
Which is now needed.

TL;DR: you can try using the file I linked to above, and I think it should fix things. Though regardless of that, please add the `[rights]` section above to your radicale.conf.

**Comment by ChasNelson1990** [comment]
Thanks. So my `radicale.conf` file has the rights type already.

I used the new `__init__.py` file. I still get the MODIFICATION_FAILED error in Thunderbird and my calendars still get reset to read only.

In terms of creating calendars - yes, I have a journal that was created in the web UI. This was just done when I was playing around one time. I've now deleted it.

After deleting, I reran etesync - no change.

It definitely seems to be dismissing the reminders that's causing the read-only state. If I turn reminders off for one of my calendars I can synchronise to my hearts content without it being set read-only.

My etesync output now includes a: `UNIQUE constraint failed: content.journal_id, content.uid
` error during PUT, which I don't remember from before.

**Comment by ChasNelson1990** [comment]
Sorry. Traceback:

>Traceback (most recent call last):
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2949, in execute_sql
    cursor.execute(sql, params or ())
sqlite3.IntegrityError: UNIQUE constraint failed: content.journal_id, content.uid
>
>During handling of the above exception, another exception occurred:
>
>Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 318, in save
    self._cache_obj.save()
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 6201, in save
    pk = self.insert(**field_dict).execute()
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 1785, in inner
    return method(self, database, *args, **kwargs)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 1856, in execute
    return self._execute(database)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2572, in _execute
    return super(Insert, self)._execute(database)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2320, in _execute
    cursor = database.execute(self)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2962, in execute
    return self.execute_sql(sql, params, commit=commit)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2956, in execute_sql
    self.commit()
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2732, in __exit__
    reraise(new_type, new_type(*exc_args), traceback)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 183, in reraise
    raise value.with_traceback(tb)
  File "/home/chas/.local/lib/python3.7/site-packages/peewee.py", line 2949, in execute_sql
    cursor.execute(sql, params or ())
peewee.IntegrityError: UNIQUE constraint failed: content.journal_id, content.uid
>
>During handling of the above exception, another exception occurred:
>
>Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 332, in __call__
    status, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 505, in _handle_request
    environ, base_prefix, path, user)
  File "/usr/lib/python3.7/site-packages/radicale/__init__.py", line 935, in do_PUT
    path, items, props)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 330, in create_collection
    self._upload_all_nonatomic(vobject_items)
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 398, in _upload_all_nonatomic
    self.upload(href, items[href])
  File "/usr/lib/python3.7/site-packages/radicale_storage_etesync/__init__.py", line 414, in upload
    etesync_item.save()
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 357, in save
    super().save()
  File "/usr/lib/python3.7/site-packages/etesync/api.py", line 321, in save
    raise exceptions.AlreadyExists(e)
etesync.exceptions.AlreadyExists: UNIQUE constraint failed: content.journal_id, content.uid

**Comment by tasn** [comment]
Could you please try with a clean slate? As in `rm -r ~/.config/etesync-dav/` and then setting up etesync-dav again? It looks like things are still bad from before. If you start with a clean slate we can see if it's just a lingering issue from before and it has been fixed or a real issue that's still there.

Thanks.

**Comment by ChasNelson1990** [comment]
Did:
`rm -r ~/.config/etesync-dav/`
`etesync-dav manage add REMOVED`
`etesync-dav`

My Thunderbird calendars synced; but when I dismiss notifications they become read-only. Note: I didn't remove the calendars from Thunderbird.

I get the same `UNIQUE constraint failed` error message.

**Comment by tasn** [comment]
Can you try also removing them from Thunderbird? Maybe it's stuck trying to do something because of a previously broken state? Are you able to add/remove events to these calendars?

Again, everything works here, including event dismissal. :|

**Comment by ChasNelson1990** [comment]
It works! Thanks for all your help.

However, I have a slight admission to make... when adding the calendars I had a vague memory that I  might have selected ics instead of caldav last time I added these...

So I added one calendar as caldav - works fine.

Added one as ics - I have the read only issue.

To make myself feel less guilty about wasting your time because I did a silly thing, I'm reminding myself that the first part of this issue might still have been a problem.

Sorry!

**Comment by tasn** [comment]
Haha :man_facepalming: 

No worries, because as you said, you actually spotted a few real issues that have now been fixed. I'll hopefully release a new version with them soon. Thanks for all the time you spent debugging this me, and I'm glad everything works now. :)
